pkg/provider/provider.go

```diff
@@ -67,22 +104,24 @@ func Select(force bool) (Provider, error) {
 	provider := ""
 	prompt := &survey.Select{
 		Message: "Select one of the following providers:",
-		Options: available,
+		Options: providers.AvailableProviders,
 	}
 	survey.AskOne(prompt, &provider, survey.WithValidator(survey.Required))
 	utils.Success("Using provider %s\n", provider)
 	return New(provider)
 }
 
 func FromManifest(man *manifest.Manifest) (Provider, error) {
-	switch man.Provider {
-	case GCP:
+	switch strings.ToUpper(man.Provider) {
+	case "GCP":
```

Comment:
I think this is going to have downstream implications, since the current values of these strings are used in configuring applications and other things.  We should certainly not modify them for safety.

Reply:
Yes it's a bit intrusive of a change

Reply:
What scares me is I'm about 99% sure it'll cause externaldns to be misconfigured, which will bork a lot of cluster setups.  But there's a ton of templating in other apps that are dependent on the existing provider casing.